pkg/kn/commands/types_test.go

```diff
@@ -0,0 +1,149 @@
+	}
+
+	if reflect.ValueOf(p.NamespaceFactory).Pointer() != reflect.ValueOf(p.CurrentNamespace).Pointer() {
+		t.Errorf("wrong function assignment detected: %v (expected) != %v (actual)", reflect.ValueOf(p.CurrentNamespace).Pointer(), reflect.ValueOf(p.NamespaceFactory).Pointer())
+	}
+}
+
+type currentNamespaceTestCase struct {
+	clientConfig  clientcmd.ClientConfig
+	namespace     string
+	expectedError error
+}
+
+func TestCurrentNamespace(t *testing.T) {
+	for i, tc := range []currentNamespaceTestCase{
+		{
+			clientcmd.NewDefaultClientConfig(clientcmdapi.Config{}, &clientcmd.ConfigOverrides{Context: clientcmdapi.Context{Namespace: "some-namespace"}}),
+			"some-namespace",
+			nil,
+		},
+	} {
+		p := &KnParams{
+			ClientConfig: tc.clientConfig}
+
+		name, err := p.CurrentNamespace()
+
+		if !reflect.DeepEqual(err, tc.expectedError) {
+			t.Errorf("%d: wrong error detected: %s (expected) != %s (actual)", i, tc.expectedError.Error(), err.Error())
+		}
+
+		if name != tc.namespace {
+			t.Errorf("%d: wrong namespace detected: %s (expected) != %s (actual)", i, tc.namespace, name)
+		}
+	}
+}
+
+type getConfigTestCase struct {
+	clientConfig      clientcmd.ClientConfig
+	expectedErrString string
+}
+
+func TestGetConfig(t *testing.T) {
+	for i, tc := range []getConfigTestCase{
+		{
+			clientcmd.NewDefaultClientConfig(clientcmdapi.Config{}, &clientcmd.ConfigOverrides{}),
+			"no configuration has been provided",
+		},
+	} {
+		p := &KnParams{
+			ClientConfig: tc.clientConfig}
+
+		_, err := p.GetConfig()
+
+		switch len(tc.expectedErrString) {
+		case 0:
+			if err != nil {
+				t.Errorf("%d: unexpected error: %s", i, err.Error())
+			}
+		default:
+			if err == nil {
+				t.Errorf("%d: wrong error detected: %s (expected) != %s (actual)", i, tc.expectedErrString, err)
+			}
+			if !strings.Contains(err.Error(), tc.expectedErrString) {
+				t.Errorf("%d: wrong error detected: %s (expected) != %s (actual)", i, tc.expectedErrString, err.Error())
+			}
+		}
+	}
+}
+
+type typeTestCase struct {
+	kubeCfgPath   string
+	explicitPath  string
+	expectedError error
+}
+
+func TestGetClientConfig(t *testing.T) {
+	multiConfigs := fmt.Sprintf("%s%s%s", "/testing/assets/kube-config-01.yml", string(os.PathListSeparator), "/testing/assets/kube-config-02.yml")
+
+	for i, tc := range []typeTestCase{
+		{
+			"",
+			clientcmd.NewDefaultClientConfigLoadingRules().ExplicitPath,
+			nil,
+		},
+		{
+			"/testing/assets/kube-config-01.yml",
+			"/testing/assets/kube-config-01.yml",
+			nil,
+		},
+		{
+			multiConfigs,
+			"",
+			errors.New(fmt.Sprintf("Config file '%s' could not be found. For configuration lookup path please use the env variable KUBECONFIG", multiConfigs)),
+		},
+	} {
+		p := &KnParams{
+			KubeCfgPath: tc.kubeCfgPath,
+		}
+
+		clientConfig, err := p.GetClientConfig()
+
+		if !reflect.DeepEqual(err, tc.expectedError) {
```

Comment:
Checking on exact error strings can be quite fragile. Better often to check whether important information is contained like the name of the config file, "not", "found", "path"). As @maximilien mentioned, we are now using the gotest.tools assertion library, so you write the test like in

```golang
assert.Assert(util.ContainsAll(err.String(), "not", "found", multiConfigs)
```